README.md

```diff
@@ -614,6 +630,20 @@ network_connections:
         - 192.168.1.1/24
 ```
 
+Configuring a wireless connection:
+
+```yaml
+network_connections:
+  - name: wlan0
+    type: wireless
+    wireless:
+      ssid: "My WPA2-PSK Network"
+      key-mgmt: "wpa-psk"
+      # recommend vault encrypting the wireless password
+      # see https://docs.ansible.com/ansible/latest/user_guide/vault.html
+      password: "p@55w0rD"
```

Comment:
Does it make sense to do it like this if we also rename 802.1x to authentication? @pcahyna opinions? 



```suggestion
    wireless:
      ssid: "My WPA2-PSK Network"
      key-mgmt: "wpa-psk"
    authentication:
      # recommend vault encrypting the wireless password
      # see https://docs.ansible.com/ansible/latest/user_guide/vault.html
      password: "p@55w0rD"
```

Maybe `key-mgmt` would also make sense as part of `authentication`...

Reply:
@tyll I grouped them this way as it corresponds to how the options are laid out in NetworkManager

I.e. all the 'wireless' options correspond to an `802-11-wireless-*` setting and the '802.1x' settings all correspond to an `802-1x.*` setting

I thought this would make the role more intuitive for users that are already familiar with nmcli and also makes the validation logic simpler.

Reply:
If we do make a generic 'authentication' key, maybe it would be good to group the settings under different subkeys for PSK and EAP? Might make the validation logic a bit less complicated.

```
authentication:
  wpa-psk:
    password: "p@55w0rD"
```

or 

```
authentication:
  wpa-eap:
      identity: myhost
      eap: tls
      private-key: /etc/pki/tls/client.key
      # recommend vault encrypting the private key password
      # see https://docs.ansible.com/ansible/latest/user_guide/vault.html
      private-key-password: "p@55w0rD"
      client-cert: /etc/pki/tls/client.pem
      ca-cert: /etc/pki/tls/cacert.pem
```

Reply:
@tyll  @pcahyna would you like me to change this?

Reply:
Thank you for your patience, @pcahyna would like to investigate regarding the possible problems with 802.1x with Jinja.
If we do 
```
authentication:
    wpa-eap:
        ...
```
It would again be a problem with `802.1x", as it would look like:

```
authentication:
    802.1x:
        ...
```

Also `wpa-eap` would also not be a valid Python identifier.
I think if we want to go with `authentication`, using something like `mode: 802.1x` would be nice:

```
authentication:
    mode: 802.1x|wpa-psk
    ....
```
Hope this would work fine regarding validation since we probably do different validation depending on the `type` key, too.

Reply:
`wpa-eap` could be called `wpa_eap` and `802.1x` could be called `ieee8021x` . Not sure whether it is better to have a key for each mode or a `mode` key with each mode as a possible value.